BCM/bcm.h

```diff
@@ -0,0 +1,140 @@
+/*
+ * bcm.h
+ *
+ * Created: 2020-03-01 9:49:57 PM
+ *  Author: AHMED_SALAMA
+ */ 
+
+
+#ifndef BCM_H_
+#define BCM_H_
+
+
+/************************************************************************/
+/*				 INCLUDES			        */
+/************************************************************************/
+
+#include "std_types.h"
+#include "bcm_cfg.h"
+
+/************************************************************************/
+/*				 DEFINES			        */
+/************************************************************************/
+
+/* BCM_ID */
+
+#define BCM_ID                                  (4)
+
+/* BCM_Dispatcher states */
+#define BCM_DISPATCHER_IDLE                     (0)
+
+/* TX states */
+#define BCM_DISPATCHER_TRANSMIT_BYTE            (1)
+#define BCM_DISPATCHER_TRANSMIT_BYTE_COMPLETE   (2)
+
+/* RX states */
+#define BCM_DISPATCHER_RECEIVE             (3)
+#define BCM_DISPATCHER_RECEIVE_COMPLETE    (4)
+
+
+/* BCM_TxDispatcher Transmit states */
+#define BCM_TX_DISPATCHER_ID_TX                 (0)
+#define BCM_TX_DISPATCHER_SIZEL_TX              (1)
+#define BCM_TX_DISPATCHER_SIZEH_TX              (2)
+#define BCM_TX_DISPATCHER_DATA_TX               (3)
+
+/* BCM TX buffer lock states */
+#define BCM_TX_BUFFER_UNLOCKED                  (0)
+#define BCM_TX_BUFFER_LOCKED                    (1)
+
+/* BCM_TxDispatcher Transmit states */
+#define BCM_RX_DISPATCHER_ID_TX                 (0)
+#define BCM_RX_DISPATCHER_SIZEL_TX              (1)
+#define BCM_RX_DISPATCHER_SIZEH_TX              (2)
+#define BCM_RX_DISPATCHER_DATA_TX               (3)
+
+/* BCM TX buffer lock states */
+#define BCM_RX_BUFFER_UNLOCKED                  (0)
+#define BCM_RX_BUFFER_LOCKED                    (1)
+
+/* BCM send state */
+#define BCM_SEND_NOT_TRIGGERED                  (0)
+#define BCM_SEND_TRIGGERED                      (1)
+
+/* BCM TX ISR Flag */
+#define BCM_TXC_FLAG_NOT_TRIGGERED              (0)
+#define BCM_TXC_FLAG_TRIGGERED                  (1)

```

Comment:
IF the user of the module needs these defines, then ok - But if some of them are implementation dependent, then we don't need them in the header file